test/common/http/http_server_properties_cache_impl_test.cc

```diff
@@ -422,6 +424,9 @@ TEST_P(HttpServerPropertiesCacheImplTest, CanonicalSuffix) {
   initialize();
   protocols_->setAlternatives(origin1, protocols1_);
 
+  protocols_->getOrCreateHttp3StatusTracker(origin1).markHttp3Broken();
```

Comment:
Can we also update `CanonicalSuffixNoMatch` test case by marking `origin1` as broken and asserting that `origin2` is not broken?

Reply:
done